sympy/solvers/ode/single.py

```diff
@@ -852,5 +852,80 @@ def _get_general_solution(self, *, simplify: bool = True):
         return [Eq(lhs, C2 + x), Eq(lhs, C2 - x)]
 
 
+class Lienard(SinglePatternODESolver):
+    r"""
+    The general form of the Lienard ODE:
+
+    .. math :: y''(x) + f(x)*y'(x) + y(x)
+
+    We solve using the fact that all linear second order homogeneous ODEs
+    can be transformed into first order ODEs of Riccati type.
+    Here we solve using the transformation as:
+
+    .. math :: b(x) = y'(x)/y(x)
+
+    .. math :: y''(x) = b'(x)*y(x) + y(x)*b(x)**2
+
+    On substituting,
+
+    .. math :: b'(x) + b(x)**2 + f(x)*b(x) + 1
+
+    So this new nonlinear first order ODE in b(x) is obtained, further we can try to find b(x)
+    by solving this ode using dsolve. Later, we can find y(x) by:
+
+    .. math :: y(x) = exp(Integral(b(x),x))
+
+
+    Examples
+    ========
+
+    >>> from sympy import Function, dsolve, Symbol
+    >>> f = Function('f')
+    >>> x = Symbol('x')
+
+    >>> eq = f(x).diff(x, x) + 2/x*f(x).diff(x) + f(x)
+    >>> sol = dsolve(eq, hint='lienard',simplify=False)
+    >>> print(sol)
+    Eq(f(x), exp(C2)/(x*sqrt(tan(C1 + x)**2 + 1)))
```

Comment:
Why do we need `simplify=False`?

Reply:
It gives `PolynomialError: non-commutative expressions are not supported` for not setting `simplify = False` when the solution of Riccati equation i.e. `sol` has `O` term like in  `eq = f(x).diff(x,2) + x*f(x).diff(x) + f(x)`. 
```
sol =  dsolve(eq)
```
Otherwise for cases like ` eq = f(x).diff(x, x) + 2/x*f(x).diff(x) + f(x)`, it works even for not imposing any conditions on `simplify`.

Kindly provide your opinion.
Thanks.

Reply:
That might be a bug. Can you find what causes that error in a way that does not use `dsolve`?

Reply:
So I printed the expression which causes the issue, 
For `eq = f(x).diff(x,2) + x*f(x).diff(x) + f(x)`
the below solution which is the answer for the ODE above, raises `PolynomialError`. 
```
C2 + exp(x**2*(-C1**2 - 1)/2 + x**4*(3*C1**2 + (-C1**2 - 1)*(6*C1**2 + 1) + 1)/24 + x**6*(-8*C1**2*(-3*C1**2 - 1) - 14*C1**2 + (-36*C1**2 - 5)*(-C1**2 - 1) + (-C1**2 - 1)*(-16*C1**2*(-3*C1**2 - 1) - 36*C1**2 + 8*(-9*C1**2 - 1)*(-C1**2 - 1) - 5) - 2)/720 + C1*x + C1*x**3*(2*C1**2 + 1)/6 + C1*x**5*(-12*C1**2 + 8*(-3*C1**2 - 1)*(-C1**2 - 1) - 5)/120 + O(x**7))
```
As you try `simplify` on it as:
```
In [1]: C1,C2=symbols('C1 C2')

In [2]: s=C2 + exp(x**2*(-C1**2 - 1)/2 + x**4*(3*C1**2 + (-C1**2 - 1)*(6*C1**2 + 1) + 1)/24 + x**6*(-8*C1**2*(-3*C1**2
   ...: - 1) - 14*C1**2 + (-36*C1**2 - 5)*(-C1**2 - 1) + (-C1**2 - 1)*(-16*C1**2*(-3*C1**2 - 1) - 36*C1**2 + 8*(-9*C1**
   ...: 2 - 1)*(-C1**2 - 1) - 5) - 2)/720 + C1*x + C1*x**3*(2*C1**2 + 1)/6 + C1*x**5*(-12*C1**2 + 8*(-3*C1**2 - 1)*(-C1
   ...: **2 - 1) - 5)/120 + O(x**7))
DEBUG: parsing of expression [(x, 1, None, None)] with symbol x
DEBUG: returned ([], [(x, 1, None, None)], 1, False)

In [3]: simplify(s)
---------------------------------------------------------------------------
PolynomialError                           Traceback (most recent call last)
~\sympy\sympy\polys\polytools.py in cancel(f, *gens, **args)
   6685             raise PolynomialError()
-> 6686         R, (F, G) = sring((p, q), *gens, **args)
   6687         if not R.ngens:

~\sympy\sympy\polys\rings.py in sring(exprs, *symbols, **options)
    163     # TODO: rewrite this so that it doesn't use expand() (see poly()).
--> 164     reps, opt = _parallel_dict_from_expr(exprs, opt)
    165

~\sympy\sympy\polys\polyutils.py in _parallel_dict_from_expr(exprs, opt)
    327     if any(expr.is_commutative is False for expr in exprs):
--> 328         raise PolynomialError('non-commutative expressions are not supported')
    329

PolynomialError: non-commutative expressions are not supported

During handling of the above exception, another exception occurred:

PolynomialError                           Traceback (most recent call last)
<ipython-input-3-d628693ded22> in <module>
----> 1 simplify(s)

~\sympy\sympy\simplify\simplify.py in simplify(expr, ratio, measure, rational, inverse, doit, **kwargs)
    634     expr = Mul(*powsimp(expr).as_content_primitive())
    635     _e = cancel(expr)
--> 636     expr1 = shorter(_e, _mexpand(_e).cancel())  # issue 6829
    637     expr2 = shorter(together(expr, deep=True), together(expr1, deep=True))
    638

~\sympy\sympy\core\expr.py in cancel(self, *gens, **args)
   3696         """See the cancel function in sympy.polys"""
   3697         from sympy.polys import cancel
-> 3698         return cancel(self, *gens, **args)
   3699
   3700     def invert(self, g, *gens, **args):

~\sympy\sympy\polys\polytools.py in cancel(f, *gens, **args)
   6692     except PolynomialError as msg:
   6693         if f.is_commutative and not f.has(Piecewise):
-> 6694             raise PolynomialError(msg)
   6695         # Handling of noncommutative and/or piecewise expressions
   6696         if f.is_Add or f.is_Mul:

PolynomialError: non-commutative expressions are not supported
```
Even though the same error occurs at a different line of the file than in `dsolve`, still it is caused by the same method `.is_commutative`, but if we manually try as:
```
In [4]: s.is_commutative
Out[4]: True
```
It returns `True`.

Reply:
A simpler way to reproduce the bug:
```
In [26]: cancel(exp(1+O(x)))
---------------------------------------------------------------------------
PolynomialError
```
Can you open a separate issue for that?

That doesn't look like it could be the cause here though. The docstring doesn't show a series solution.

Reply:
Sure. #20973 
Thanks.